src/utils/file-io/deserialize/deserializeProject.ts

```diff
@@ -0,0 +1,51 @@
+import { logger } from "utils/common/helpers";
+import { CustomStore } from "../zarrStores";
+import { openGroup } from "zarr";
+import { getAttr } from "./helpers";
+import semver from "semver";
+import { deserializeProject_v1 } from "./v1/deserializeProject_v1";
+import { dataConverter_v1v2 } from "utils/file-io/converters/dataConverter_v1v2";
+import { deserializeProject_v2 } from "./v2/deserializeProject_v2";
+import { LoadCB } from "../types";
+
+export const deserializeProject = async (
+  fileStore: CustomStore,
+  loadCb: LoadCB
+) => {
+  process.env.REACT_APP_LOG_LEVEL === "1" &&
+    logger(`starting deserialization of ${fileStore.rootName}`);
+
+  const rootGroup = await openGroup(fileStore, fileStore.rootName, "r");
+
+  const piximiVersionRaw = (await getAttr(rootGroup, "version")) as string;
+  const piximiVersion = semver.clean(piximiVersionRaw);
+
+  if (!semver.valid(piximiVersion) || semver.lt(piximiVersion!, "0.1.0")) {
+    throw Error(`File version ${piximiVersion} is unsupported.`);
+  } else if (semver.eq(piximiVersion!, "0.1.0")) {
+    const { project, classifier, data, segmenter } =
+      await deserializeProject_v1(fileStore, loadCb);
+    const { kinds, categories, things } = dataConverter_v1v2({
```

Comment:
When we are in version `0.x.y`, we are allowed to introduce substantial changes between values of `x`. Hence it makes sense to change our project file format between `0.1.y` and `0.2.y`.
However, when we are in version `z.x.y`, we should not have substantial changes unless we are incrementing the value of `z`.

Once we eventually do reach `v1` and then sometime after that `v2`, we may want to have `dataConverter_v1v2` and `deserializeProject_v1` and `deserializeProject_v2` available to us.
For the `v0` series of updates, we may use something like `dataConverter_v0102` or `vz1` or something to denote that we are in the "special" `0` series.

Reply:
Should we change the version checking now? or leave it as it. The names have been changes from vx->v0x in the lates commit